src/Entity/SnapshotManager.php

```diff
@@ -99,16 +100,17 @@ public function enableSnapshots(array $snapshots, ?\DateTime $date = null)
         }
 
         $this->getEntityManager()->flush();
-        // @todo: strange sql and low-level pdo usage: use dql or qb
-        $sql = sprintf(
-            "UPDATE %s SET publication_date_end = '%s' WHERE id NOT IN(%s) AND page_id IN (%s) and publication_date_end IS NULL",
-            $this->getTableName(),
-            $date->format('Y-m-d H:i:s'),
-            implode(',', $snapshotIds),
-            implode(',', $pageIds)
-        );
 
-        $this->getConnection()->query($sql);
+        $qb = $this->getRepository()->createQueryBuilder('s');
+        $q = $qb->update()
+            ->set('s.publicationDateEnd', ':date_end')
+            ->where($qb->expr()->notIn('s.id', $snapshotIds))
+            ->andWhere($qb->expr()->in('s.page_id', $pageIds))
+            ->andWhere($qb->expr()->isNull('s.publicationDateEnd'))
+            ->setParameter('date_end', $date, Types::DATETIME_MUTABLE)
```

Comment:
why this type `Types::DATETIME_MUTABLE`?

Reply:
in my attempt to do it with Mock tests, it was failing because "datetime" doesn't have a id field to map or something

thats why i said i need another tests where i can do functional tests

Reply:
seems it worked without, the mocking was incomplete

Reply:
Also that why:
>  QueryBuilderSetParameter: To improve performance set explicit type for objects

https://github.com/psalm/psalm-plugin-symfony/issues/158